From 6279dce005f42dfa84108e5be9c5f572b9245c29 Mon Sep 17 00:00:00 2001
From: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
Date: Fri, 18 Mar 2016 11:06:56 +0700
Subject: [PATCH 4/5] qtmultimedia: qmlvideo add fullscreen display support

run qmlvideo with --fullscreen
display in fullscreen can give better performance

Signed-off-by: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
Signed-off-by: Huynh Thanh Hung <hung.huynh.wz@renesas.com>
---
 examples/multimedia/video/qmlvideo/main.cpp | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/examples/multimedia/video/qmlvideo/main.cpp b/examples/multimedia/video/qmlvideo/main.cpp
index cc4437f8c..e2a615f59 100644
--- a/examples/multimedia/video/qmlvideo/main.cpp
+++ b/examples/multimedia/video/qmlvideo/main.cpp
@@ -79,6 +79,9 @@ int main(int argc, char *argv[])
 #endif
             else if ("-url" == arg) {
                 sourceIsUrl = true;
+            } 
+            else if ("--fullscreen" == arg) {
+		//Do nothing
             } else {
                 qtTrace() << "Option" << arg << "ignored";
             }
@@ -130,8 +133,13 @@ int main(int argc, char *argv[])
 
     QMetaObject::invokeMethod(rootObject, "init");
 
-    viewer.setMinimumSize(QSize(640, 360));
-    viewer.show();
+    const QString lowerArgument = QString::fromLatin1(argv[1]).toLower();
+    if (lowerArgument == QLatin1String("--fullscreen")) {
+        viewer.showFullScreen();
+    } else {
+        viewer.setMinimumSize(QSize(640, 360));
+        viewer.show();
+    }
 
     return app.exec();
 }
-- 
2.17.1

